Realm/RLMRealm.mm

```diff
@@ -373,9 +373,24 @@ + (instancetype)realmWithConfiguration:(RLMRealmConfiguration *)configuration er
                         schema = [RLMSchema.sharedSchema copy];
                         realm->_realm->update_schema(schema.objectStoreCopy, config.schema_version);
                     }
+                } else {
+                    realm->_realm->update_schema(schema.objectStoreCopy, config.schema_version);
                 }
 
                 RLMRealmSetSchemaAndAlign(realm, schema);
+            } catch (SchemaMismatchException::SchemaMismatchException const& exception) {
+                if (configuration.deleteRealmIfMigrationNeeded) {
+                    NSError *deletiionError = nil;
```

Comment:
I would like this option (`deleteRealmIfMigrationNeeded`)!
I found it seemed like a typo to me.